cmd/tools/grafana/grafana.go

```diff
@@ -523,45 +544,43 @@ func createFolder(opts *options) error {
 		return errors.New("server response: " + status)
 	}
 
-	opts.folderId = int64(result["id"].(float64))
-	//opts.folderId = strconv.FormatFloat(result["id"].(float64), 'f', 0, 32)
-	//opts.folderId = result["id"].(string)
-	opts.folderUid = result["uid"].(string)
+	opts.folder[i].folderId = int64(result["id"].(float64))
+	opts.folder[i].folderUid = result["uid"].(string)
 
 	return nil
 }
 
-func checkFolder(opts *options) (bool, error) {
+func checkFolder(opts *options) ([]bool, error) {
 
 	result, status, code, err := sendRequestArray(opts, "GET", "/api/folders?limit=1000", nil)
 
 	if err != nil {
-		return false, err
+		return []bool{false, false}, err
 	}
 
 	if code != 200 {
-		return false, errors.New("server response: " + status)
+		return []bool{false, false}, errors.New("server response: " + status)
 	}
 
 	if result == nil || len(result) == 0 {
-		return false, nil
+		return []bool{false, false}, nil
 	}
 
+	cmodeFolder := folderExist(result, &opts.folder[0])
+	nonCmodeFolder := folderExist(result, &opts.folder[1])
+
+	return []bool{cmodeFolder, nonCmodeFolder}, nil
+}
+
+func folderExist(result []map[string]interface{}, folder *Folder) bool {
 	for _, x := range result {
-		//elem := x.(map[string]interface{})
-		if x["title"].(string) == opts.folder {
-			//opts.folderId = strconv.FormatFloat(x["id"].(float64), 'f', 0, 32)
-			opts.folderId = int64(x["id"].(float64))
-			opts.folderUid = x["uid"].(string)
-
-			// DEBUG
-			//fmt.Println("FOUND FOLDER!")
-			//x.Print(0)
-			return true, nil
+		if x["title"].(string) == folder.folderName {
```

Comment:
if title is missing from the map, the cast will cause this code to panic, might be better to check if title exists first using the ok idiom

Reply:
looks little odd, but still added checks for all 3 and it can pass only when dashboard have all 3 title, id and uid.